crates/virtual-node/src/lib.rs

```diff
@@ -101,54 +108,99 @@ impl VirtualNode {
+    /// Create a new `VirtualDomElement`.
     ///
-    /// The DomUpdater uses this to look up elements and see if they're still in the page. If not
-    /// the refernce that we maintain to their closure will be dropped, thus freeing the Closure's
-    /// memory.
-    pub closures: HashMap<u32, Vec<DynClosure>>,
-}
+    /// This is the variant that is contained inside `VirtualNode::Element`. It
+    /// can be turned into a `VirtualNode` using the from/into.
+    pub fn element_variant<S: Into<String>>(tag: S) -> VirtualNodeElement {
+        VirtualNodeElement::new(tag)
+    }
 
-impl Deref for CreatedElement {
-    type Target = Element;
+    /// Create a new `VirtualDomText`.
+    ///
+    /// This is the variant that is contained inside `VirtualNode::Text`. It
+    /// can be turned into a `VirtualNode` using the from/into.
+    pub fn text_variant<S: Into<String>>(text: S) -> VirtualNodeText {
+        VirtualNodeText { text: text.into() }
+    }
 
-    fn deref(&self) -> &Self::Target {
-        &self.element
+    /// Cast into a `VirtualNodeElement` reference, if this is an `Element` variant.
+    pub fn as_element_variant_ref(&self) -> Option<&VirtualNodeElement> {
+        match self {
+            VirtualNode::Element(ref element_node) => Some(element_node),
+            _ => None,
+        }
     }
-}
 
-fn create_unique_identifier() -> u32 {
-    let mut elem_unique_id = ELEM_UNIQUE_ID.lock().unwrap();
+    /// Cast into a mutable `VirtualNodeElement` reference, if this is an `Element` variant.
+    pub fn as_element_variant_ref_mut(&mut self) -> Option<&mut VirtualNodeElement> {
+        match self {
+            VirtualNode::Element(ref mut element_node) => Some(element_node),
+            _ => None,
+        }
+    }
 
-    *elem_unique_id += 1;
+    /// Cast into a `VirtualNodeText` reference, if this is a `Text` variant.
+    pub fn as_text_variant_ref(&self) -> Option<&VirtualNodeText> {
+        match self {
+            VirtualNode::Text(ref text_node) => Some(text_node),
+            _ => None,
+        }
+    }
 
-    *elem_unique_id
+    /// Cast into a mutable `VirtualNodeText` reference, if this is a `Text` variant.
+    pub fn as_text_variant_ref_mut(&mut self) -> Option<&mut VirtualNodeText> {
+        match self {
+            VirtualNode::Text(ref mut text_node) => Some(text_node),
+            _ => None,
+        }
+    }
+
+    /// Create and return a `CreatedNode` instance (containing a DOM `Node`
+    /// together with potentially related closures) for this virtual node.
+    pub fn create_dom_node(&self) -> CreatedNode<Node> {
+        match self {
+            VirtualNode::Text(text_node) => CreatedNode::without_closures(text_node.create_text_node()),
+            VirtualNode::Element(element_node) => element_node.create_element_node().into(),
+        }
+    }
+
+    /// Whether or not this `VirtualNode` is representing a `Text` node
```

Comment:
Can we remove this now? I'm not super concerned about stability into `v1.0` and the number of people using `.is_text_node()` directly vs. using the macro is probably approximately 0 ha

Reply:
Done. I planned to remove it, but then forgot :slightly_smiling_face: